libraries/abstractions/transport/secure_sockets/transport_secure_sockets.c

```diff
@@ -80,93 +80,89 @@ static TransportSocketStatus_t establishConnect( NetworkContext_t * pNetworkCont
  */
 static int32_t tlsSetup( const SocketsConfig_t * pSocketsConfig,
                          Socket_t tcpSocket,
-                         const uint8_t * pHostName,
+                         const char * pHostName,
                          size_t hostnameLength );
 
 
 /*-----------------------------------------------------------*/
 
-int32_t SecureSocketsTransport_Send( NetworkContext_t * pNetworkContext,
+int32_t SecureSocketsTransport_Send( const NetworkContext_t * pNetworkContext,
                                      const void * pMessage,
                                      size_t bytesToSend )
 {
-    size_t bytesSent = 0;
-    int32_t TransportSocketStatus = SOCKETS_ERROR_NONE;
-    Socket_t xSocket = SOCKETS_INVALID_SOCKET;
+    int32_t bytesSent = 0;
+    int32_t transportSocketStatus = SOCKETS_ERROR_NONE;
 
     if( ( pMessage == NULL ) ||
         ( bytesToSend <= ( size_t ) 0 ) ||
         ( pNetworkContext == NULL ) ||
         ( pNetworkContext->pContext == NULL ) )
     {
-        LogError( ( "TransportSendSecureSockets bad parameters pMessage %p, bytesToSend %d,  pvContext %p",
+        LogError( ( "Invalid parameter: pMessage=%p, bytesToSend=%d,  pContext=%p, pNetworkContext=%p",
                     pMessage, bytesToSend, pNetworkContext->pContext ) );
-        bytesSent = ( size_t ) SOCKETS_EINVAL;
+        bytesSent = SOCKETS_EINVAL;
```

Comment:
`pNetworkContext->pContext` can cause a Seg Fault if `pNetworkContext` is NULL. Thus, it would be better to test NULL check of `pNetworkContext->pContext` in a separate if-check

Reply:
fixed in b02e976